pkg/scalers/couchdb_scaler.go

```diff
@@ -0,0 +1,247 @@
+	} else {
+		return nil, "", fmt.Errorf("no query given")
+	}
+
+	if val, ok := config.TriggerMetadata["queryValue"]; ok {
+		queryValue, err := strconv.ParseInt(val, 10, 64)
+		if err != nil {
+			return nil, "", fmt.Errorf("failed to convert %v to int, because of %v", queryValue, err.Error())
+		}
+		meta.queryValue = queryValue
+	} else {
+		return nil, "", fmt.Errorf("no queryValue given")
+	}
+
+	meta.activationQueryValue = 0
+	if val, ok := config.TriggerMetadata["activationQueryValue"]; ok {
+		activationQueryValue, err := strconv.ParseInt(val, 10, 64)
+		if err != nil {
+			return nil, "", fmt.Errorf("failed to convert %v to int, because of %v", activationQueryValue, err.Error())
+		}
+		meta.activationQueryValue = activationQueryValue
+	}
+
+	meta.dbName, err = GetFromAuthOrMeta(config, "dbName")
+	if err != nil {
+		return nil, "", err
+	}
+
+	// Resolve connectionString
+	switch {
+	case config.AuthParams["connectionString"] != "":
+		meta.connectionString = config.AuthParams["connectionString"]
+	case config.TriggerMetadata["connectionStringFromEnv"] != "":
+		meta.connectionString = config.ResolvedEnv[config.TriggerMetadata["connectionStringFromEnv"]]
+	default:
+		meta.connectionString = ""
+		meta.host, err = GetFromAuthOrMeta(config, "host")
+		if err != nil {
+			return nil, "", err
+		}
+
+		meta.port, err = GetFromAuthOrMeta(config, "port")
+		if err != nil {
+			return nil, "", err
+		}
+
+		meta.username, err = GetFromAuthOrMeta(config, "username")
+		if err != nil {
+			return nil, "", err
+		}
+
+		if config.AuthParams["password"] != "" {
+			meta.password = config.AuthParams["password"]
+		} else if config.TriggerMetadata["passwordFromEnv"] != "" {
+			meta.password = config.ResolvedEnv[config.TriggerMetadata["passwordFromEnv"]]
+		}
+		if len(meta.password) == 0 {
+			return nil, "", fmt.Errorf("no password given")
+		}
+	}
+
+	if meta.connectionString != "" {
+		connStr = meta.connectionString
+	} else {
+		// Build connection str
+		addr := fmt.Sprintf("%s:%s", meta.host, meta.port)
+		// auth := fmt.Sprintf("%s:%s", meta.username, meta.password)
+		connStr = "http://" + addr
+	}
+	if val, ok := config.TriggerMetadata["metricName"]; ok {
+		meta.metricName = kedautil.NormalizeString(fmt.Sprintf("couchdb-%s", val))
+	}
+	meta.scalerIndex = config.ScalerIndex
+	return &meta, connStr, nil
+}
+
+func NewCouchDBScaler(ctx context.Context, config *ScalerConfig) (Scaler, error) {
+	metricType, err := GetMetricTargetType(config)
+	if err != nil {
+		return nil, fmt.Errorf("error getting scaler metric type: %s", err)
+	}
+
+	meta, connStr, err := parseCouchDBMetadata(config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parsing couchDB metadata, because of %v", err)
+	}
+
+	client, err := kivik.New("couch", connStr)
+	if err != nil {
+		fmt.Println("Couldn't connect to couchdb: ", err)
+		return nil, fmt.Errorf("%v", err)
+	}
+
+	err = client.Authenticate(context.TODO(), couchdb.BasicAuth("admin", meta.password))
```

Comment:
Do we need to use password always? I mean, if user has provided the connection string, from the code I think that we don't require user/password, so this line it isn't needed (password will be empty indeed). What about username?
Apart from that, use the context received as parameter:
```suggestion
	err = client.Authenticate(ctx, couchdb.BasicAuth("admin", meta.password))
```

Reply:
Yes, password is needed